src/opm/io/eclipse/ESmry.cpp

```diff
@@ -219,40 +232,81 @@ ESmry::ESmry(const std::string &filename, bool loadBaseRunData)
-        unsmry.loadData();
-
-        std::vector<EclFile::EclEntry> list1 = unsmry.getList();
-
-        // 2 or 3 arrays pr time step.
-        //   If timestep is a report step:  MINISTEP, PARAMS and SEQHDR
-        //   else : MINISTEP and PARAMS
+        rootN = formattedVect[n] ? smspecFile.substr(0, smspecFile.size() - 7) : smspecFile.substr(0, smspecFile.size() - 6);
+        
+        // check if multiple or unified result files should be used 
+        // to import data, no information in smspec file regarding this
+        // if both unified and non-unified files exists, will use most recent based on 
+        // time stamp
 
-        // if summary file starts with a SEQHDR, this will be ignored
+        std::string unsmryFile = formattedVect[n] ? rootN + "FUNSMRY" : rootN + "UNSMRY";
+        
+        bool use_unified = boost::filesystem::exists(unsmryFile);
+        
+        std::vector<std::string> resultsFileList;
+        std::vector<std::string> multFileList = checkForMultipleResultFiles(rootN, formattedVect[n]);
+        
+        if ((!use_unified) && (multFileList.size()==0)){
+            throw std::runtime_error("neigther unified or non-unified result files found");
+        } else if ((use_unified) && (multFileList.size()>0)){
+            auto time_multiple = boost::filesystem::last_write_time(multFileList.back());
+            auto time_unified = boost::filesystem::last_write_time(unsmryFile);
+            
+            if (time_multiple > time_unified){
+                resultsFileList=multFileList;
+            } else {
+                resultsFileList.push_back(unsmryFile);
+            }
+            
+        } else if (use_unified){
+            resultsFileList.push_back(unsmryFile);
+        } else {
+            resultsFileList=multFileList;
+        }
+        
+        // make array list with reference to source files (unifed or non unified)
+        
+        std::vector<std::tuple<std::string, std::string, int>> arraySourceList;
 
-        int i = 0;
+        for (std::string fileName : resultsFileList){
+            EclFile unsmry(fileName);
+                
+            std::vector<EclFile::EclEntry> arrayList = unsmry.getList();
 
-        if (std::get<0>(list1[0]) == "SEQHDR") {
-            i = 1;
+            for (size_t nn = 0; nn < arrayList.size(); nn++){
+                std::tuple<std::string, std::string, int> t1 = std::make_tuple(std::get<0>(arrayList[nn]), fileName, static_cast<int>(nn));
+                arraySourceList.push_back(t1);
+            }
         }
 
-        while  (i < static_cast<int>(list1.size())){
+        // loop through arrays and extract symmary data from result files, arrays PARAMS
+        //
+        //    2 or 3 arrays pr time step.
+        //       If timestep is a report step:  MINISTEP, PARAMS and SEQHDR
+        //       else : MINISTEP and PARAMS
+        
+        
+        int i = std::get<0>(arraySourceList[0]) == "SEQHDR" ? 1 : 0 ;
```

Comment:
Why do use `int` here - and then `static_cast<int>(std::size_t)` many times further down - can't you just use `std::size_t` all the way?

Reply:
Agree, I just need to getting used to use size_t. Thanks.